proto/viam/app/data/v1/data.proto

```diff
@@ -173,3 +185,54 @@ message DeleteBinaryDataByIDsRequest {
 message DeleteBinaryDataByIDsResponse {
   uint64 deleted_count = 1;
 }
+
+// Annotation specifies the ways in which we can associate annotations with data, e.g. with a text tag or image bounding box
+message Annotation {
+  string tag = 1;
+}
+
+// AddAnnotationsToBinaryDataByFileIDsRequest requests adding annotations to the files when file IDs are provided
+message AddAnnotationsToBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  repeated Annotation annotations = 2;
+}
+
+// AddAnnotationsToBinaryDataByFileIDsResponse return IDs of added annotations in order to pass it to RemoveAnnotations
+message AddAnnotationsToBinaryDataByFileIDsResponse {
+  repeated string annotation_ids = 1;
+}
+
+// AddAnnotationsToBinaryDataByFilterRequest requests adding annotations to the files when a filter is provided
+message AddAnnotationsToBinaryDataByFilterRequest {
+  Filter filter = 1;
+  repeated Annotation annotations = 2;
+}
+
+// AddAnnotationsToBinaryDataByFilterResponse return IDs of added annotations in order to pass it to RemoveAnnotations
+message AddAnnotationsToBinaryDataByFilterResponse {
+  repeated string annotation_ids = 1;
+}
+
+// RemoveAnnotationsFromBinaryDataByFilterRequest requests removing the annotations from the file when file IDs are provided
+message RemoveAnnotationsFromBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  // If annotation_ids is empty, remove all annotations for the specified files.
+  repeated string annotation_ids = 2;
+}
+
+// RemoveAnnotationsFromBinaryDataByFilterResponse returns the number of binary files whose annotations were removed when file IDs are provided
```

Comment:
nit: think can remove "when file IDs are provided" throughout

And supernit: perhaps "returns the number of binary files which had annotations removed" here and below?

Reply:
done!